zebra-network/src/peer/client.rs

```diff
@@ -168,6 +168,14 @@ impl<T: std::fmt::Debug> MustUseOneshotSender<T> {
             .unwrap_or_else(
                 || panic!("called is_canceled() after using oneshot sender: oneshot must be used exactly once: {:?}", self))
     }
+
+    /// Returns true if the application is shutting down.
+    ///
+    /// Returns false otherwise.
+    pub fn is_shutting_down(&self) -> bool {
+        use std::sync::atomic::Ordering;
+        crate::IS_SHUTTING_DOWN.load(Ordering::Relaxed)
```

Comment:
We might need `SeqCst` ordering when we start sharing `IS_SHUTTING_DOWN` between crates using an `Arc`. `Arc`s also use atomics, and we need the write to happen within the `Arc` critical section.

```suggestion
        // ## Correctness:
        //
        // Since we're shutting down, and this is a one-time operation,
        // performance is not important. So we use the strongest memory
        // ordering.
        // https://doc.rust-lang.org/nomicon/atomics.html#sequentially-consistent
        crate::IS_SHUTTING_DOWN.load(Ordering::SeqCst)
```